### PR TITLE
Test fetch cancelling

### PR DESCRIPTION
**Pull request by michaelnisi**
This adds a test showing how for cancelled fetches, the result handler is only called when `.fetchIgnoringCacheData` is the selected cache policy. Not exactly the same, but related to #514. Does it resolve it?

Although, this PR is suggesting a fix—the flag in line 92 toggles back to the previous state—improving documentation for cancelled fetches would help. But the question remains:

For cancelled fetches, should `resultHandler: ((GraphQLResult<Query.Data>?, Error?) -> Void)?` be called:

1) always with `(nil, nil)`
2) always with something else
3) never
4) sometimes (well documented)

🤔

**Do not merge**  but please discuss.

### CODE REVIEW

File: Sources/Apollo/ApolloClient.swift [hunk 1]
```diff
@@ -87,7 +87,9 @@ public class ApolloClient {
   func _fetch<Query: GraphQLQuery>(query: Query, cachePolicy: CachePolicy, context: UnsafeMutableRawPointer? = nil, queue: DispatchQueue, resultHandler: OperationResultHandler<Query>?) -> Cancellable {
     // If we don't have to go through the cache, there is no need to create an operation 
     // and we can return a network task directly
-    if cachePolicy == .fetchIgnoringCacheData {
+    //
+    // But without an operation, we lose the ability to cancel properly.
+    if false, cachePolicy == .fetchIgnoringCacheData {
```
Comment (RolandasRazma) [inline Sources/Apollo/ApolloClient.swift]: ? this is hardcoded false

Comment (michaelnisi) [reply Sources/Apollo/ApolloClient.swift]: It’s the flag I’ve mentioned in my comment above. Removing the `false` reverts the code to the previous *indeterministic* behaviour, in which it doesn't pass the newly added test. With this PR I’m trying to find out how cancelled fetches should deal with the result handler. Once that’s defined, the tests can be updated, and eventually the implementation adjusted. Or we can decide to keep the current behaviour and just improve documentation.

**Comment by designatednerd** [comment]
@michaelnisi Please check out `.fetchIgnoringCacheCompletely`, as well as [`EmptyCancellable`](https://github.com/apollographql/apollo-ios/blob/master/Sources/Apollo/Cancellable.swift#L16) - do either of these help your case here?

**Comment by michaelnisi** [comment]
Thanks

**Pull request closed without merging**